packages/eas-cli/src/analytics/commands/common.ts

```diff
@@ -0,0 +1,24 @@
+import { Analytics, Event } from './events';
+
+export type TrackingContext = Record<string, string | number | boolean>;
+
+export async function withAnalyticsAsync<Result>(
+  fn: () => Promise<Result>,
+  analytics: {
+    successEvent: Event;
+    failureEvent: Event;
+    trackingCtx: TrackingContext;
+  }
+): Promise<Result> {
+  try {
```

Comment:
might be unnecessary, but do we also want an 'attempt' stat here? cc @animeeples 
should be the one-liner here, then adding in the new event types at the call sites

Reply:
im open to it, but would prefer to do it in a separate PR :)